test/test_spark.py

```diff
@@ -1669,3 +1669,46 @@ def test_to_list(self):
 
         with pytest.raises(ValueError):
             util.to_list(['item1', 'item2'], 4)
+
+    def test_dbfs_local_store(self):
+        import h5py
+        import io
+
+        from tensorflow import keras
+
+        # test Store.create will not automatically create DBFSLocalStore
+        local_store = Store.create("/dbfs/tmp/test_local_dir")
+        assert isinstance(local_store, LocalStore)
+
+        # test get_checkpoint_filename suffix
+        dbfs_store = DBFSLocalStore("/dbfs/tmp/test_dbfs_dir")
+        dbfs_ckpt_name = dbfs_store.get_checkpoint_filename()
+        assert dbfs_ckpt_name.endswith(".tf")
+
+        # prepare for testing read_serialized_keras_model
+        model = keras.Sequential([keras.Input((32,)), keras.layers.Dense(1)])
+
+        def deserialize_keras_model(serialized_model):
+            model_bytes = codec.loads_base64(serialized_model)
+            bio = io.BytesIO(model_bytes)
+            with h5py.File(bio, 'r') as f:
+                return keras.models.load_model(f)
+
+        # test dbfs_store.read_serialized_keras_model
+        dbfs_ckpt_path = dbfs_store.prefix_path + "/" + dbfs_ckpt_name
+        model.save(dbfs_ckpt_path)
+        serialized_model_dbfs = dbfs_store.read_serialized_keras_model(dbfs_ckpt_path)
+        reconstructed_model_dbfs = deserialize_keras_model(serialized_model_dbfs)
+        assert reconstructed_model_dbfs.get_config() == model.get_config()
+        assert reconstructed_model_dbfs.get_weights() == model.get_weights()
+
+        # test local_store.read_serialized_keras_model
+        local_ckpt_path = local_store.prefix_path + "/" + local_store.get_checkpoint_filename()
+        model.save(local_ckpt_path)
```

Comment:
For TF 1.15.0, SavedModel format is not selected by adding the ".tf" suffix. https://github.com/tensorflow/docs/blob/920057b528427ac647b8dcb93d898fce76507d3d/site/en/r1/guide/saved_model.md#build-and-load-a-savedmodel The syntax for saving in SavedModel is quite different from the current one.
Manual test on databricks with tf downgraded to 1.15 also fails with the h5py error.

Reply:
https://dogfood.staging.cloud.databricks.com/#notebook/29305617/command/29305633